### PR TITLE
[Test] Make test_comparison_utils.py device-generic

### PR DESCRIPTION
**Pull request by vishalgoyal316**
> Refactor `test/test_comparison_utils.py` so `torch._assert_tensor_metadata` tests run on every available backend, not only implicit CPU.

### Summary

`TestComparisonUtils` tests metadata assertion for export/compile guards. The op compares sizes, strides, dtype, device, and layout -- no kernel numerics -- so it is safe to run device-generically.

Previously six tests used `torch.tensor([0.5])` without `device=`, so they only ran once on CPU. `test_assert_device` was gated with `@unittest.skipIf(not torch.cuda.is_available())` and hardcoded `cpu` vs `cuda`, even though the test only checks device *type* mismatch in metadata.

### Changes

- Add `device` parameter to all seven test methods
- Add `_mismatched_device` helper (aligned with `get_another_device` in `test_tensor_creation_ops.py`)
- Remove unnecessary CUDA skip from `test_assert_device`; `SkipTest` only when no second device type exists
- Add `instantiate_device_type_tests(..., except_for=("meta",), allow_mps=True, allow_xpu=True)`
- Add `hw_classification = HardwareClassification.ACCELERATOR` to `TestComparisonUtils` per hardware classification guidelines (https://github.com/pytorch/pytorch/pull/186918).
 
### Test plan

```bash
python test/test_comparison_utils.py
```

14 tests passed locally (7 methods x CPU + CUDA). MPS/XPU variants are generated when those backends are available.

<sub>Authored with an AI assistant.</sub>

### CODE REVIEW

**Comment by vishalgoyal316** [comment]
cc: @fffrog @KarhouTam @RiyaP2508 @mansiag05

File: test/test_comparison_utils.py [hunk 1]
```diff
@@ -4,49 +4,70 @@
...
+        t = torch.tensor([0.5], device=device)
+        wrong_device = _mismatched_device(device)
+        if wrong_device is None:
+            raise unittest.SkipTest(
+                "need at least two device types for device mismatch test"
+            )
 
         with self.assertRaises(RuntimeError):
-            torch._assert_tensor_metadata(t, device="cuda")
+            torch._assert_tensor_metadata(t, device=wrong_device)
```
Comment (fffrog) [inline test/test_comparison_utils.py]:
```Python
    def test_assert_device(self, device):
        t = torch.tensor([0.5], device="cpu")

        with self.assertRaises(RuntimeError):
            torch._assert_tensor_metadata(t, device=device)
```

Comment (vishalgoyal316) [reply test/test_comparison_utils.py]: Hi @fffrog, thanks for the suggestion, and it looks better approach.
I will update the PR accordingly.

**Comment by jbschlosser** [comment]
@claude review this PR and call out any places where the change to be device-agnostic is incomplete

File: test/test_comparison_utils.py [hunk 2]
```diff
@@ -1,52 +1,58 @@
...
+        t = torch.tensor([0.5], device=device)
 
         with self.assertRaises(RuntimeError):
             torch._assert_tensor_metadata(t, layout=torch.sparse_coo)
 
 
+instantiate_device_type_tests(
+    TestComparisonUtils,
+    globals(),
+    except_for=("meta", "cpu"),
```
Comment (jbschlosser) [inline test/test_comparison_utils.py]:
```suggestion
```

We don't want to drop CPU coverage :p

Comment (vishalgoyal316) [reply test/test_comparison_utils.py]: Thanks!, updated the code!

**Comment by KarhouTam** [comment]
Seems red CIs are related. Please fix them first, thanks @vishalgoyal316 .

**Comment by vishalgoyal316** [comment]
Hello @jbschlosser, @KarhouTam, thanks for comments, following changes to fix the CI failure:
  
Fixed test_assert_device by adding a _mismatched_device(device) helper that finds an available device type different from the one under test at runtime. The tensor is now created on device (not hardcoded "cpu"), and the mismatch is asserted against wrong_device. 
Added except_for=("meta",) since _assert_tensor_metadata skips device checks for meta tensors by design. 
All 14 tests pass (7 CPU + 7 CUDA).

File: test/test_comparison_utils.py [hunk 3]
```diff
@@ -1,52 +1,67 @@
...
 
+def _mismatched_device(device):
+    device = torch.device(device)
+    if device.type != "cpu":
+        return "cpu"
+    acc = torch.accelerator.current_accelerator(True)
+    return acc.type if acc is not None else None
+
+
 class TestComparisonUtils(TestCase):
```
Comment (Niran814804102) [inline test/test_comparison_utils.py:21]: Acccording to https://github.com/pytorch/pytorch/pull/189376, perhaps the label hw_classification = HardwareClassification should be added here, eg. hw_classification = HardwareClassification.ACCELERATOR

Comment (vishalgoyal316) [reply test/test_comparison_utils.py:21]: Hello @Niran814804102, thanks for the feedback, I have update PR for the same, please help to review again.

Comment (Niran814804102) [reply test/test_comparison_utils.py:21]: Thanks for your adoptation.

**Review by Niran814804102 — APPROVED** [review]

**Comment by vishalgoyal316** [comment]
Hello @jbschlosser, @fffrog, PR is updated with review comments addressed, could you please help to review this PR again, thanks!

**Review by mansiag05 — APPROVED** [review]
